lib/handle_callback.ts

```diff
@@ -80,14 +92,17 @@ export async function handleCallback(
     ...options?.cookieOptions,
   };
   setCookie(response.headers, cookie);
+
+  const sessionData = await tokenCallback(tokens.accessToken);
+  if (sessionData === null || sessionData === undefined) {
+    throw new Error("tokenCallback() must resolve to a non-nullable value");
+  }
+
   await setSiteSession(
     sessionId,
+    sessionData,
     cookie.maxAge ? cookie.maxAge * SECOND : undefined,
   );
 
-  return {
-    response,
-    sessionId,
-    tokens,
-  };
+  return response;
```

Comment:
I think it may be useful to still return the tokens from this function. Either that pass the whole token object to `tokenCallback`? I feel like some people may want the ability to ensure sessions are still valid and be able to refresh the access token in the cases that it is not.

This used to be built into this module but got removed, which I understand the reasoning for, but I feel like some (or many idk) use-cases would require checking/refreshing the session

Reply:
It used to store the access/refresh tokens in KV and would automatically refresh the session but that was removed some time back. I remember fixing some bugs in that area but it was removed soon after